pkg/agent/agent_windows.go

```diff
@@ -32,3 +40,127 @@ func (i *Initializer) setupExternalConnectivity() error {
 	}
 	return nil
 }
+
+// prepareHostNetwork creates HNS Network for containers.
+func (i *Initializer) prepareHostNetwork() error {
+	i.nodeConfig.UplinkNetConfig = new(config.AdapterNetConfig)
+	// If the HNS Network already exists, return immediately.
+	hnsNetwork, err := hcsshim.GetHNSNetworkByName(util.LocalHNSNetwork)
+	if err == nil {
+		// Save the uplink adapter name to check if the OVS uplink port has been created in prepareOVSBridge stage.
+		i.nodeConfig.UplinkNetConfig.Name = hnsNetwork.NetworkAdapterName
+		return nil
+	}
+	if _, ok := err.(hcsshim.NetworkNotFoundError); !ok {
+		return err
+	}
+	// Get uplink network configuration.
+	_, adapter, err := util.GetIPNetDeviceFromIP(i.nodeConfig.NodeIPAddr.IP)
+	if err != nil {
+		return err
+	}
+	i.nodeConfig.UplinkNetConfig.Name = adapter.Name
+	i.nodeConfig.UplinkNetConfig.MAC = adapter.HardwareAddr
+	i.nodeConfig.UplinkNetConfig.IP = i.nodeConfig.NodeIPAddr
+	i.nodeConfig.UplinkNetConfig.Index = adapter.Index
+	defaultGW, err := util.GetDefaultGatewayByInterfaceIndex(adapter.Index)
+	if err != nil {
+		return err
+	}
+	i.nodeConfig.UplinkNetConfig.Gateway = defaultGW
+	dnsServers, err := util.GetDNServersByInterfaceIndex(adapter.Index)
+	if err != nil {
+		return err
+	}
+	i.nodeConfig.UplinkNetConfig.DNSServers = dnsServers
+	// Create HNS network.
+	return util.PrepareHNSNetwork(i.nodeConfig.PodCIDR, i.nodeConfig.NodeIPAddr, adapter)
+}
+
+// cleanupHostNetwork deletes HNS Network created by Antrea.
+func (i *Initializer) cleanupHostNetwork() error {
+	return util.DeleteHNSNetwork(util.LocalHNSNetwork)
+}
+
+// prepareOVSBridge adds local port and uplink to ovs bridge.
+func (i *Initializer) prepareOVSBridge() error {
+	hnsNetwork, err := hcsshim.GetHNSNetworkByName(util.LocalHNSNetwork)
+	if err != nil {
+		return err
+	}
+
+	// Set datapathID of OVS bridge.
+	// If no datapathID configured explicitly, the reconfiguration operation will change OVS bridge datapathID
+	// and break the OpenFlow channel.
+	// The length of datapathID is 64 bits, the lower 48-bits are for a MAC address, while the upper 16-bits are
+	// implementer-defined. Antrea uses "0x0000" for the upper 16-bits.
+	datapathID := strings.Replace(hnsNetwork.SourceMac, ":", "", -1)
+	datapathID = "0000" + datapathID
+	if err = i.ovsBridgeClient.SetDatapathID(datapathID); err != nil {
+		klog.Errorf("Failed to set datapath_id %s: %v", datapathID, err)
+		return err
+	}
+
+	// Create local port.
+	brName := i.ovsBridgeClient.GetBridgeName()
+	if _, err := i.ovsBridgeClient.GetOFPort(brName); err == nil {
+		klog.Infof("OVS bridge local port %s already exists, skip the configuration", brName)
+	} else {
+		if _, err := i.ovsBridgeClient.CreateInternalPort(brName, config.AutoAssignedOFPort, nil); err != nil {
+			return err
+		}
+	}
+
+	// If uplink is already exists, return.
+	uplinkNetConfig := i.nodeConfig.UplinkNetConfig
+	uplink := uplinkNetConfig.Name
+	if uplink == "" {
```

Comment:
In what cases the uplink is ""? It looks always have a value. Then maybe this check is not needed.

Reply:
- On linux platform, the value of uplink will be "".
- On windows it will be the name of uplink interface.

Reply:
But it is agent_windows.go, I don't think this method is called on Linux.

Reply:
> But it is agent_windows.go, I don't think this method is called on Linux.

Sorry, you are correct. My last comment is for the value check in initInterfaceStore.

I think this check can be remvoed.